build.gradle

```diff
@@ -579,6 +579,22 @@ gradle.projectsEvaluated {
       }
     }
   }
+  project.allprojects.forEach { outer ->
+    project.allprojects.forEach { inner ->
```

Comment:
This is O(N^2)....Instead we can have one loop over the projects and build a map with the group/name as the key. Also, I think this needs to be in a projectsEvaluated or it won't capture when the group is changed during configuration, since this code is running during root configuration?

Reply:
It is already in a projectsEvaluated, but I added it to an existing one thus it doesn't show in the diff. 
We wrap this individually for better visibility.

I'm not convinces the impact in performance will be noticeable, as the number of projects is not arbitrarily large nor can it grow significantly. I think the gain in readability is worth it.

Reply:
I don't think this is more readable. I had to stare at this a while to understand what was going on. A single loop over all the projects, collecting them by their group/name seems much more transparent to me:

```
Map coordsToProject = []
project.allprojects.forEach { p ->
  String coords = "${p.group}:${p.name}"
  if (false == coords.putIfAbsent(coords, p)) {
    throw new GradleException(a message)
  }
}
```

I would also add a comment explaining the problem and linking to the gradle bug before the loop.